### DISCUSSION TITLE
Command "code" not found

### DISCUSSION THREAD

**Discussion by Mystic8b** [microsoft/WSL#6356] [General] [answered]
# Environment

```none
Windows build number: [ Microsoft Windows [Version 10.0.19042.685 ]
Your Distribution version: [ Debian Release: 20.10 ]
Whether the issue is on WSL 2 and/or WSL 1: [ Linux version 4.19.128-microsoft-standard (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Tue Jun 23 12:58:10 UTC 2020 ]
```

# Steps to reproduce

- install vscode
- pwsh: `wsl --set-default-version 2`
- Install Debian 20.04 from Win Store
- Upgrade Debian to 20.10
- Installed packages:

  - mc
  - wget
  - bzip2
  - htop
  - ncdu
  - git
  - tcpdump
  - traceroute
  - dnsutils
  - net-tools
  - neofetch
  - lolcat
  - build-essential
  - ansible
  - python3-pip
  - python3-dev
  - nload
  - zsh + omz + p10k

- run `code .`

#  Expected behavior

vscode opens

# Actual behavior

```
~ $ zsh
❯ code .
Command 'code' not found, did you mean:
  command 'tcode' from deb emboss (6.6.0+dfsg-7ubuntu2)
  command 'node' from deb nodejs (12.18.2~dfsg-1ubuntu2)
  command 'ode' from deb plotutils (2.6-11)
  command 'cdde' from deb cdde (0.3.1-1build1)
  command 'cde' from deb cde (0.1+git9-g551e54d-1.1build1)
Try: sudo apt install <deb name>
❯ echo $PATH
/usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin /usr/games /usr/local/games /mnt/c/Program Files/Common Files/Oracle/Java/javapath /mnt/c/Program Files (x86)/Common Files/Oracle/Java/javapath /mnt/c/Windows/system32 /mnt/c/Windows /mnt/c/Windows/System32/Wbem /mnt/c/Windows/System32/WindowsPowerShell/v1.0/ /mnt/c/Windows/System32/OpenSSH/ /mnt/c/Program Files (x86)/Common Files/Acronis/SnapAPI/ /mnt/c/Program Files/Git/cmd /mnt/c/Users/Mannet/AppData/Local/Microsoft/WindowsApps

❯ bash
~$ code .
Command 'code' not found, did you mean:
  command 'node' from deb nodejs (12.18.2~dfsg-1ubuntu2)
  command 'cde' from deb cde (0.1+git9-g551e54d-1.1build1)
  command 'ode' from deb plotutils (2.6-11)
  command 'cdde' from deb cdde (0.3.1-1build1)
  command 'tcode' from deb emboss (6.6.0+dfsg-7ubuntu2)
Try: sudo apt install <deb name>
~$ echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Program Files/Common Files/Oracle/Java/javapath:/mnt/c/Program Files (x86)/Common Files/Oracle/Java/javapath:/mnt/c/Windows/system32:/mnt/c/Windows:/mnt/c/Windows/System32/Wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0/:/mnt/c/Windows/System32/OpenSSH/:/mnt/c/Program Files (x86)/Common Files/Acronis/SnapAPI/:/mnt/c/Program Files/Git/cmd:/mnt/c/Users/Mannet/AppData/Local/Microsoft/WindowsApps
```

ps This is not the first issue with this problem, but apparently it still hasn't been resolved.
ps I understand that the problem is the absence of code in the path, but what is the reason for this?

**Comment by therealkenc** [comment]
>  I understand that the problem is the absence of code in the path

Yours is different than the [canonical reason](https://github.com/microsoft/WSL/issues/5779), but confirm `/etc/profile` isn't clobbering your `$PATH` just as a sanity check. It really can't be, because you've got other Windows `%PATH%`s in the list, but not `code`.

Code usually lives in this path:
```
/mnt/c/Users/there/AppData/Local/Programs/Microsoft VS Code/bin/code
```

That is not in your list. Questions:

Can you run `code` with the absolute path (you need to quote it to run).

What is the output of:
```
$ cmd.exe /c "echo %PATH%"
```

And finally, does your problem reproduce on an unadulterated Ubuntu without installing `zsh` related materials and `bash` as the default shell.

**Comment by Mystic8b** [comment]
> confirm `/etc/profile` isn't clobbering your `$PATH` just as a sanity check

```
❯ cat /etc/profile
# /etc/profile: system-wide .profile file for the Bourne shell (sh(1))
# and Bourne compatible shells (bash(1), ksh(1), ash(1), ...).

if [ "${PS1-}" ]; then
  if [ "${BASH-}" ] && [ "$BASH" != "/bin/sh" ]; then
    # The file bash.bashrc already sets the default PS1.
    # PS1='\h:\w\$ '
    if [ -f /etc/bash.bashrc ]; then
      . /etc/bash.bashrc
    fi
  else
    if [ "`id -u`" -eq 0 ]; then
      PS1='# '
    else
      PS1='$ '
    fi
  fi
fi

if [ -d /etc/profile.d ]; then
  for i in /etc/profile.d/*.sh; do
    if [ -r $i ]; then
      . $i
    fi
  done
  unset i
fi
```

> Can you run `code` with the absolute path (you need to quote it to run).

Opens:
![Снимок экрана 2020-12-25 133128](https://user-images.githubusercontent.com/35394377/103132445-97847f80-46b5-11eb-996c-f37fef59b42e.jpg)

> What is the output

```
❯ cmd.exe /c "echo %PATH%"
"\\wsl$\Ubuntu\home\mannet"
Указанный путь был использован при запуске CMD.EXE в качестве текущей папки.
CMD.EXE не поддерживает пути UNC. По умолчанию выбрана системная папка Windows.
C:\Program Files\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C
:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files (x86)\Common Files\Acronis
\SnapAPI\;C:\Program Files\Git\cmd;C:\Users\Mannet\AppData\Local\Microsoft\WindowsApps
```

It's like:
```
CMD.EXE was started with the above path as the current directory.
UNC paths are not supported.  Defaulting to Windows directory.
```

> And finally, does your problem reproduce on an unadulterated Ubuntu without installing `zsh` related materials and `bash` as the default shell.

Yes, the problem still exists
![Снимок экрана 2020-12-25 131422](https://user-images.githubusercontent.com/35394377/103131827-5f7c3d00-46b3-11eb-97b8-d155b793cecc.jpg)

**Comment by therealkenc** [comment, marked as answer]
> `❯ cmd.exe /c "echo %PATH%"`
`[...]`
`C:\Program Files\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files (x86)\Common Files\Acronis\SnapAPI\;C:\Program Files\Git\cmd;C:\Users\Mannet\AppData\Local\Microsoft\WindowsApps`

Thanks for the follow up and screeshots. That it starts given the full path is good news. 

Your `code` application is not in your Windows `%PATH%`. Normally that gets set up when you install Code in Windows via the installer. In your case it either didn't get set up, or got clobbered along the way. You can add the path manually in control panel (being careful it takes, or reboot to be sure). Or reinstalling Code will probably work. Looks like:

![image](https://user-images.githubusercontent.com/7049700/103228979-ea914900-48e6-11eb-9ee7-98af6a1c4dd6.png)

**Comment by Mystic8b** [comment]
Oooh, this is my bad. I apologize, apparently I unchecked the "add to path" checkbox myself ...
Thank you for your time!
